### DISCUSSION TITLE
Initial deployment of Bicep registry modules (Azure DevOps Pipelines)

### DISCUSSION THREAD

**Discussion by kevball2** [Azure/ResourceModules#2503] [Q&A] [answered]
Reading through the documentation for CI and Publishing I did not find a section for deploying the initial version of the bicep modules to a registry. 

The Publish-ModuleToPrivateBicepRegistry.ps1 and Get-ModuleToPrivateBicepRegistry.ps1 files do not check for deployment of an initial version of the module. Since there is no change found by Git no modules are deployed. Is there a step that I am missing for this? 

Deploying modules through Azure DevOps

**Comment by eriqua** [comment, marked as answer]
Hey @kevball2, sorry for the late reply.

What you describe looks related to issue #2153 which was closed a couple of weeks ago through PR #2410 .

The publishing logic has been updated to publish not only when module code changes are detected, but also when no module is found in the target location.

Although changes are not released yet, they are in the main branch. You can test them already in your environment. Let us know if that solves!

**Reply by AlexanderSehr** [reply]
I can only echo what @eriqua wrote. The logic should work now in a combination of the functions
- [`Publish-ModuleToPrivateBicepRegistry.ps1`](https://github.com/Azure/ResourceModules/blob/main/utilities/pipelines/resourcePublish/Publish-ModuleToPrivateBicepRegistry.ps1) and
- [`Get-ModulesMissingFromPrivateBicepRegistry.ps1`](https://github.com/Azure/ResourceModules/blob/main/utilities/pipelines/resourcePublish/Get-ModulesMissingFromPrivateBicepRegistry.ps1)

**Comment by kevball2** [comment]
Thanks to both of you for the information. Looks like I'll have to look at that PR, unfortunately I have a client who did not want to use public copy of the repo so I will need to merge it in myself :(

**Reply by AlexanderSehr** [reply]
All good @kevball2, if you need any support ler me/us know and we can figure it out :)
